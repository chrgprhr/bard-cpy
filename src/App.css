.App {
  text-align: center;
  background-color: #f0f4f9;
  height: 100%;
  width: 100%;
  overflow: hidden;
}

.header {
  justify-content: space-between;
  padding: 8px;
  height: 8%;
}

.headerLeft {
  display: flex;
  justify-content: space-around;
  padding-right: 30px;
}

.headerMenu {
  padding: 12px;
  margin: 0 5px;
}

.content {
  height: 85%;
  overflow: hidden;
  display: flex;
}

.fadein {
  position: relative;
  animation: fadeIn 0.4s 1;
  animation-fill-mode: forwards;
}

@keyframes fadeIn {
  0%   { 
    top: 15px;
    display: none;
  }
  100% { top: 0px; }
}

.fadeinleft {
  position: relative;
  animation: fadeInLeft 0.4s 1;
  animation-fill-mode: forwards;
}

@keyframes fadeInLeft {
  0%   { 
    left: -15px;
    display: none;
  }
  100% { 
    left: 0px;
  }
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.a-center {
  display: flex;
  align-items: center;
}

.j-center {
  display: flex;
  align-items: center;
  justify-content: center;
}

.j-btw {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.cursor-pointer {
  cursor: pointer;

  &:hover {
    background-color: #dde3ea;
    border-radius: 50px;
  }
}
