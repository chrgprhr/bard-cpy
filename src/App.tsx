import { Header } from "./Header";
import { BardViewContextProvider } from "./BardViewContext";
import { Menu } from "./Menu";
import { NewChatView } from "./NewChatView";

import "./App.css";

const View = () => {
  return (
    <>
      <Header />
      <div className="content">
        <Menu />
        <NewChatView />
      </div>
    </>
  );
};

function App() {
  return (
    <div className="App">
      <BardViewContextProvider>
        <View />
      </BardViewContextProvider>
    </div>
  );
}

export default App;
