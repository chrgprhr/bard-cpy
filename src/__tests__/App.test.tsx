import { render, act, fireEvent, waitFor } from "@testing-library/react";
import App from "../App";

jest.mock("@mui/material-next/Button", () => () => {
  return (
    <div>
      <button></button>
    </div>
  );
});

jest.mock("react-markdown", () => ({ children }: { children: any }) => {
  return <div>{children}</div>;
});

describe("dummy test", () => {
  test("adds 1 + 2 to equal 3", () => {
    expect(1 + 2).toBe(3);
  });
});

describe("Main page tests", () => {
  it("renders and matches the snapshot", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot("main-page");
  });

  it("should preill the query input when clicked on the prompt tiles", async () => {
    const { getByText, getByDisplayValue } = render(<App />);

    const dummyTileOption = getByText("Create a script");
    expect(dummyTileOption).toBeTruthy();

    act(() => {
      fireEvent.click(dummyTileOption!);
    });

    await waitFor(() => {
      expect(getByDisplayValue("Write me a script for")).toBeTruthy();
    });
  });

  it("should display the mock response when query is typed and submitted", () => {
    const { container, getByPlaceholderText } = render(<App />);

    const queryInput = getByPlaceholderText("Enter a prompt here");
    expect(queryInput).toBeTruthy();

    act(() => {
      fireEvent.change(queryInput, {
        target: { value: "dummytypeduserquery" },
      });
    });

    act(() => {
      fireEvent.click(container.getElementsByClassName("send-icon")[0]);
    });

    expect(container).toMatchSnapshot("mock response on query submit");
  });

  it("should display the recent queries tags after submit", () => {
    const { container, getByText } = render(<App />);

    const dummyTileOption = getByText("Create a script");

    act(() => {
      fireEvent.click(dummyTileOption!);
    });

    act(() => {
      fireEvent.click(container.getElementsByClassName("send-icon")[0]);
    });

    // Click on the menu
    act(() => {
      fireEvent.click(
        container.getElementsByClassName("headerMenu cursor-pointer")[0]
      );
    });

    expect(getByText("Recent")).toBeDefined();
    expect(getByText("Script writing")).toBeDefined();
  });

  it("should handle no response case", () => {
    const { container, getByText, getByDisplayValue } = render(<App />);

    const dummyTileOption = getByText("No Response");

    act(() => {
      fireEvent.click(dummyTileOption!);
    });

    act(() => {
      fireEvent.click(container.getElementsByClassName("send-icon")[0]);
    });

    expect(container).toMatchSnapshot("no response snapshot");

    act(() => {
      fireEvent.click(
        container.getElementsByClassName("activeQueryQuestionEdit")[0]
      );
    });

    expect(
      getByDisplayValue("This query will give you no response")
    ).toBeTruthy();
  });
});
