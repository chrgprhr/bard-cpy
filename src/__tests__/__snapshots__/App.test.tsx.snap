// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page tests renders and matches the snapshot: main-page 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="header a-center"
    >
      <div
        class="headerLeft"
      >
        <div
          class="headerMenu cursor-pointer"
        >
          <svg
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </div>
        <div
          class="a-center bardLabel"
        >
          <a
            aria-label="Google Bard"
            class="a-center"
            href="/"
            title="Google Bard"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="https://www.gstatic.com/lamda/images/OGB_badge_en_v1_light_d9fefe2c502b5bf2d6b1.svg"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="chat-view "
      >
        <div
          class="chat-view-child tiles-section"
        >
          <div
            class="top-row j-btw"
          >
            <div>
              <img
                _ngcontent-ng-c3877805786=""
                alt="bard-cpy-img"
                height="36"
                src="https://www.gstatic.com/lamda/images/sparkle_resting_v2_1ff6f6a71f2d298b1a31.gif"
                width="36"
              />
            </div>
            <button
              aria-label="delete"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="RefreshIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="head1"
          >
            Hello again
          </div>
          <div
            class="head2"
          >
            Tell me whatâ€™s on your mind, or pick a suggestion.
          </div>
          <div
            class="prompt-tiles j-btw"
          >
            <div
              class="prompt-tile"
            >
              <div
                class="prompt-tile-heading heading-0"
              >
                Creativity
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Write a poem
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Craft a story
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Generate code
              </div>
            </div>
            <div
              class="prompt-tile"
            >
              <div
                class="prompt-tile-heading heading-1"
              >
                Information
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Summarize a topic
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Translate a language
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Answer a question
              </div>
            </div>
            <div
              class="prompt-tile"
            >
              <div
                class="prompt-tile-heading heading-2"
              >
                Expression
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Write a song
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Create a script
              </div>
              <div
                class="prompt-option cursor-pointer fadein "
              >
                Compose an email
              </div>
            </div>
          </div>
        </div>
        <div
          class="chat-view-child input-section"
        >
          <div
            class="a-center"
          >
            <input
              class="bard-input"
              placeholder="Enter a prompt here"
              value=""
            />
            <div
              class="send-icon j-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-1db085k-MuiSvgIcon-root"
                data-testid="SendIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </div>
          </div>
          <p
            class="input-warning"
          >
            Bard may display inaccurate info, including about people, so double-check its responses. Your privacy & Bard
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main page tests should display the mock response when query is typed and submitted: mock response on query submit 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="header a-center"
    >
      <div
        class="headerLeft"
      >
        <div
          class="headerMenu cursor-pointer"
        >
          <svg
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </div>
        <div
          class="a-center bardLabel"
        >
          <a
            aria-label="Google Bard"
            class="a-center"
            href="/"
            title="Google Bard"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="https://www.gstatic.com/lamda/images/OGB_badge_en_v1_light_d9fefe2c502b5bf2d6b1.svg"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="chat-view "
      >
        <div
          class="chat-view-child tiles-section"
        >
          <div
            class="responseWindow fadein"
          >
            <div
              class="question"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                />
              </svg>
              <div
                class="activeQueryQuestion"
              >
                dummytypeduserquery
              </div>
            </div>
            <div
              class="qResponse"
            >
              <div>
                <img
                  _ngcontent-ng-c3877805786=""
                  alt="bard-cpy-img"
                  height="30"
                  src="https://www.gstatic.com/lamda/images/sparkle_resting_v2_1ff6f6a71f2d298b1a31.gif"
                  width="30"
                />
              </div>
              <div
                class="activeResponse"
              >
                <div>
                  This is a sample response with *emphasis* and **strong importance**.

&gt; A block quote with ~strikethrough~ and a URL: https://reactjs.org.

* Lists
* [ ] todo
* [x] done

A table:

| a | b |
| - | - |

ðŸ‘‰ Changes are re-rendered as you type.

ðŸ‘ˆ Try writing some markdown on the left.

## Overview

- Follows [CommonMark](https://commonmark.org)
- Optionally follows [GitHub Flavored Markdown](https://github.github.com/gfm/)
- Has a lot of plugins

## GitHub flavored markdown (GFM)

For GFM, you can *also* use a plugin:
[remark-gfm](https://github.com/remarkjs/react-markdown#use).
It adds support for GitHub-specific extensions to the language:
tables, strikethrough, tasklists, and literal URLs.

These features **do not work by default**.
ðŸ‘† Use the toggle above to add the plugin.

| Feature    | Support              |
| ---------: | :------------------- |
| CommonMark | 100%                 |
| GFM        | 100% w/ remark-gfm |

~~strikethrough~~

* [ ] task list
* [x] checked item

https://example.com

## More info?

Much more info is available in the
[readme on GitHub](https://github.com/remarkjs/react-markdown)!

***

A component by [Espen Hovlandsdal](https://espen.codes/)


                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chat-view-child input-section"
        >
          <div
            class="a-center"
          >
            <input
              class="bard-input"
              placeholder="Enter a prompt here"
              value="dummytypeduserquery"
            />
            <div
              class="send-icon j-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="SendIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </div>
          </div>
          <p
            class="input-warning"
          >
            Bard may display inaccurate info, including about people, so double-check its responses. Your privacy & Bard
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
